### ISSUE TITLE
Seed based world generation

### ISSUE THREAD

**Issue by jabelch** [akintu/akintu#14]
Seed based customization:

Instead of having a single seed for the entire content generation, I envision more control.  You might like the overall layout but want to have the items change.  In reality, the map seed would be the mother seed with the others mostly dependent on it.  You could keep the others the same, but on a new map it would change them enough that it would feel unique.
1. Map seed: I see this as being the main look of the map.  The overall layout of towns and dungeons is determined here. A separate attribute might be used for size.
2. Monster seed: The position, type and difficulty of monsters is chosen here with a separate attribute for the number of monsters.
3. Item seed: The position and types of items is chosen here with a separate attribute for the number of items.
4. Shop seed
   ... and any others we might come up with.  I created a wiki page with this same text, but would like some input.

@jzar @blackvegetable @basepi

**Comment by basepi** [comment]
I think we should have two random number generators with separate seeds.  One for the world and everything in it except for monsters.  The other is for monsters.  Since monsters will respawn, etc, we will not expose this seed to the user, so even though a user might re-create the same world, the monsters in it would be different.  But we definitely only need two seeds -- one for everything we want the user to be able to recreate, one for everything else.  Then we just get a new random number whenever we need to generate anything.  No point in having multiple seeds -- in fact, this would only decrease the randomization -- when you want randomization in your program, you generally want a single, global random number generator, as this will "guarantee" the most randomness.  I'm not sure how clear I'm being, I can explain more in person if necessary.

**Comment by BlackVegetable** [comment]
Colton is right about the math.  I also agree with him on having two seeds with one hidden.

**Comment by basepi** [comment]
We also talked more after class, and world generation will probably rotate around treasure chests -- we spawn a treasure chest, with a certain quality number tied to it.  We'd then generate a locked door and maybe a boss to protect it, generate a floor plan around it, and add additional dungeon levels above it.  This would all be done with our user-exposed seed.  The non-boss monsters would then be generated on the fly, using our respawn algorithms.  The item in the chest would be generated when the chest was opened, based on the quality number.

@Jzar and I also talked about how these main treasures should always be useful to your class -- random chests you find in the world will generate items completely randomly -- however, the main chests at the bottom of 5-level dungeons will always spawn something useful to the player.  They will also be individually lootable by each member of a party.  Any objections to this model?
